src/rdkafka_request.c

```diff
@@ -2431,6 +2434,13 @@ rd_kafka_error_t *rd_kafka_ListGroupsRequest(rd_kafka_broker_t *rkb,
                 rd_kafka_buf_finalize_arraycnt(rkbuf, of_GroupsArrayCnt, i);
         }
 
+        if (ApiVersion >= 5) {
+                rd_kafka_buf_write_arraycnt(rkbuf,types_cnt);
+                for (i = 0; i < types_cnt; i++) {
+                        rd_kafka_buf_write_str(rkbuf, types[i], -1);
+                }
+        }
+
```

Comment:
There is a `rd_kafka_buf_write_tags_empty` missing here as per the protocol.

```
ListGroups Request (Version: 5) => [states_filter] [types_filter] TAG_BUFFER 
  states_filter => COMPACT_STRING
  types_filter => COMPACT_STRING
```

Reply:
Removed, as it's written by `rd_kafka_buf_finalize` inside `rd_kafka_broker_buf_enq_replyq`